File: Assets/Plugins/SDK/AdRevenue/Runtime/Native/Android/Proxy/AdRevenueAdapterProxy.cs
#if UNITY_ANDROID
using UnityEngine;

namespace Io.AppMetrica.AdRevenueAdapter.Native.Android.Proxy {
    internal static class AdRevenueAdapterProxy {
        private static readonly AndroidJavaClass NativeClass = new AndroidJavaClass("io.appmetrica.adrevenueadapter.plugin.unity.AdRevenueAdapterProxy");
        
        public static void ReportAdRevenue(string adRevenue) {
            NativeClass.CallStatic("reportAdRevenue", adRevenue);
        }
    }
}
#endif

File: Assets/Plugins/SDK/AdRevenue/Runtime/Plugins/Android/AdRevenueAdapterProxy.java
package io.appmetrica.adrevenueadapter.plugin.unity;

import android.util.Log;
import androidx.annotation.NonNull;
import org.json.JSONException;
import io.appmetrica.analytics.ModulesFacade;

public class AdRevenueAdapterProxy {

    public static void reportAdRevenue(@NonNull String adRevenue) {
        try {
            ModulesFacade.reportAdRevenue(AdRevenueDeserializer.fromJsonString(adRevenue), true);
        } catch (JSONException e) {
            Log.e("AdRevenueAdapterUnity", "Failed to report AdRevenue. Data was parsed with error", e);
        }
    }

}
